code/game/objects/items/tanks/tanks.dm

```diff
@@ -339,4 +339,41 @@
 	if(gas_change)
 		air_contents.garbage_collect()
 
+/obj/item/tank/attack_hand(mob/living/carbon/human/user)
+
+	var/celsius_temperature = KELVIN_TO_CELSIUS(air_contents.temperature)
+
+	// make it burn hands unless you're wearing heat insulated gloves or have the RESISTHEAT/RESISTHEATHANDS traits (recycled comment as it still fits)
+	if(celsius_temperature >= 50)
+		var/protection = FALSE
+		var/mob/living/carbon/human/human = user
+		if(human.gloves)
+			var/obj/item/clothing/gloves/gloves = human.gloves
+			if(gloves.max_heat_protection_temperature)
+				protection = (gloves.max_heat_protection_temperature > 360)
```

Comment:
protection = TRUE and merge the conditional into the if() above perhaps?

Reply:
this allow different levels of heat protection (max_heat_protection_temperature could be less than 360 in theory)